x/gov/handler_test.go

```diff
@@ -1020,24 +998,59 @@ func TestTally(t *testing.T) {
 			},
 			ExpResult: Proposal_Accepted,
 			PostChecks: func(t *testing.T, db weave.KVStore) {
-				elect, err := NewElectorateBucket().GetElectorate(db, weavetest.SequenceID(1))
+				_, obj, err := NewElectorateBucket().GetLatestVersion(db, weavetest.SequenceID(1))
 				if err != nil {
 					t.Fatalf("unexpected error: %+v", err)
 				}
+				elect, _ := asElectorate(obj)
+				if exp, got := uint32(3), elect.Version; exp != got {
+					t.Errorf("expected %v but got %v", exp, got)
+				}
 				got := elect.Electors
-				exp := []Elector{{alice, 10}, {bobby, 10}}
+				exp := []Elector{{alice, 10}, {bobby, 10}, {charlie, 2}}
 				sortByAddress(exp)
 				if !reflect.DeepEqual(exp, got) {
 					t.Errorf("expected %v but got %v", exp, got)
 				}
-				if exp, got := uint64(20), elect.TotalElectorateWeight; exp != got {
+				if exp, got := uint64(22), elect.TotalElectorateWeight; exp != got {
 					t.Errorf("expected %v but got %v", exp, got)
 				}
 			},
 		},
+		"Does not complete tally when executor fails": {
```

Comment:
These are cases where I think the table-driven tests approach breaks down.
When the case is really a bunch of code to execute and the for loop is relatively sparse.

(I prefer when these are just data values. Fair enough. Maybe these can be broken into simple (not table) tests and be a bit more clear)

Reply:
Agree, they became a bit too big and complex

Reply:
Agree with separate PR.
Each of these two are good improvements, can merge them in and clean up tests later